assets/controllers/copy_to_clipboard_controller.js

```
@@ -0,0 +1,75 @@
+    button.classList.add(defaultColorClass);
+
+    if (textToCopy) {
+      navigator.clipboard.writeText(textToCopy)
+        .then(() => {
+          this.dispatch("success", { detail: { text: textToCopy } });
+
+          setTimeout(() => {
+            button.classList.remove(defaultColorClass);
+            button.classList.add(successColorClass);
+            copyIcon.classList.remove(active);
+            copyIcon.classList.add(unactive);
+            successIcon.classList.remove(unactive);
+            successIcon.classList.add(active);
+            
+            setTimeout(() => {
+              button.classList.remove(successColorClass);
+              button.classList.add(defaultColorClass);
+              copyIcon.classList.remove(unactive);
+              copyIcon.classList.add(active);
+              successIcon.classList.remove(active);
+              successIcon.classList.add(unactive);
+            }, 1000);
+          }, 1000);
+        })
+        .catch(err => {
+          console.error("Erreur lors de la copie dans le presse-papiers :", err);
+          this.dispatch("error", { detail: { error: err } });
+
+          setTimeout(() => {
+            button.classList.remove(defaultColorClass);
+            button.classList.add(errorColorClass);
+            copyIcon.classList.remove(active);
+            copyIcon.classList.add(unactive);
+            successIcon.classList.remove(unactive);
+            successIcon.classList.add(active);
+            
+            setTimeout(() => {
+              button.classList.remove(errorColorClass);
+              button.classList.add(defaultColorClass);
+              copyIcon.classList.remove(unactive);
+              copyIcon.classList.add(active);
+              successIcon.classList.remove(active);
+              successIcon.classList.add(unactive); 
+            }, 1000);
+          }, 1000);
+        });
+    } else {
+      console.warn("Aucun texte à copier");test
```

This `test` token at the end of this line should throw an error.